test/e2e/controller_test.go

```diff
@@ -137,19 +140,19 @@ var _ = Describe("Combination controller", func() {
 				g.Expect(conditionReasons).To(ContainElement(v1alpha1.ReasonProcessed))
 				g.Expect(retrievedCombination.Status.Evaluations).To(ContainElements(expectedUpdatedEvaluations))
 
-				return err
-			}).Should(BeZero())
+				return nil
+			}).Should(Succeed())
 		})
 
-		It("should have the combo.ReferencedTemplate label correctly applied on the combination", func() {
+		It("should have the ReferencedTemplate label correctly applied on the combination", func() {
 			Eventually(func(g Gomega) error {
 				var retrievedCombination v1alpha1.Combination
 				err := kubeclient.Get(ctx, types.NamespacedName{Name: validCombinationCRCopy.Name}, &retrievedCombination)
 
-				g.Expect(retrievedCombination.Labels).To(HaveKeyWithValue("combo.ReferencedTemplate", validTemplateCRCopy.Name))
+				g.Expect(retrievedCombination.Labels).To(HaveKeyWithValue(controller.ReferencedTemplateLabel, validTemplateCRCopy.Name))
```

Comment:
nit: I'd personally like to avoid mixing gomega (or testify) assertions within an Eventually block, and just handle the error (or whatever check we're interested in performing) directly using conditional checks and appropriately returning.